File: app/src/main/java/torille/fi/lurkforreddit/subreddit/SubredditPresenterModule.java
package torille.fi.lurkforreddit.subreddit;

import dagger.Module;
import dagger.Provides;
import torille.fi.lurkforreddit.data.RedditRepository;
import torille.fi.lurkforreddit.data.models.view.Subreddit;

/**
 * This is a Dagger module. We use this to pass in the View dependency to the
 * {@link SubredditPresenter}.
 */
@Module
public class SubredditPresenterModule {


    private final Subreddit mSubreddit;

    public SubredditPresenterModule(Subreddit subreddit) {
        mSubreddit = subreddit;
    }

    @Provides
    public SubredditContract.Presenter<SubredditContract.View> provideSubredditPresenter(RedditRepository redditRepository,
                                                                                         Subreddit subreddit) {
        return new SubredditPresenter(redditRepository, subreddit);
    }

    @Provides
    Subreddit provideSubreddit() {
        return mSubreddit;
    }

}

File: app/src/main/java/torille/fi/lurkforreddit/subreddit/SubredditPresenterModule.kt
package torille.fi.lurkforreddit.subreddit

import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import torille.fi.lurkforreddit.di.scope.FragmentScoped

@Module
abstract class SubredditPresenterModule {

    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun subredditFragment(): SubredditFragment

    @FragmentScoped
    @Binds
    abstract fun subredditPresenter(presenter: SubredditPresenter): SubredditContract.Presenter

}
